sdk/src/main/java/com/kin/ecosystem/recovery/BackupAndRestoreImpl.java

```diff
@@ -127,6 +114,28 @@ public void onFailure(BackupException exception) {
 		});
 	}
 
+	private BackupManager getNewBackupManager() {
+		return new BackupManager(activity, blockchainSource.getKeyStoreProvider());
+	}
+
+	private void switchAccount(int accountIndex, @NonNull final BackupAndRestoreCallback backupCallback) {
+		accountManager.switchAccount(accountIndex, new KinCallback<Boolean>() {
+			@Override
+			public void onResponse(Boolean response) {
+				release();
+				backupManager = getNewBackupManager(); //update the backupManager
+				eventLogger.send(RestoreWalletCompleted.create());
+				backupCallback.onSuccess();
+			}
+
+			@Override
+			public void onFailure(KinEcosystemException exception) {
+				backupCallback.onFailure(new BackupAndRestoreException(RESTORE_SWITCH_ACCOUNT_FAILED,
+					exception != null ? exception.getMessage() : "Switch account failed", exception));
```

Comment:
move string to final static member.

Reply:
@drEfrati ^^^

Reply:
It was like this before but will do it now.